### DISCUSSION TITLE
Is it possible to avoid a running dag run that would be affected if the dag file is changed?

### DISCUSSION THREAD

**Discussion by shawn-ann** [apache/airflow#41421] [Q&A]
For example, I create a dag file with a dag id of "dag-001" containing a task A and B. A will take 30 minutes to execute, I run the dag file(first run), and while it's running I update the file and add a new task after B, and then run it again(second run).

My expectation is that the first run should only run tasks A and B, but it will also run C, even though I didn't add C at startup.

In short, for running dag runs, any changes to the dag file should not affect it.

**Comment by potiuk** [comment]
See https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-63%3A+DAG+Versioning -> this is planned in Airflow 3
